statistics/handle/statscache.go

```diff
@@ -15,15 +15,271 @@ package handle
 
 import (
 	"encoding/binary"
+	"errors"
 	"sync"
+	"time"
 
+	"github.com/pingcap/badger/cache"
 	"github.com/pingcap/tidb/statistics"
 	"github.com/pingcap/tidb/util/kvcache"
 	"github.com/pingcap/tidb/util/memory"
 )
 
-// statsCache caches table statistics.
-type statsCache struct {
+// StatsCache a interface can LookUp Update
+type StatsCache interface {
+	Lookup(id int64) (*statistics.Table, bool)
+	Update(tables []*statistics.Table, deletedIDs []int64, newVersion uint64)
+	GetVersion() uint64
+	InitStatsCache(tables map[int64]*statistics.Table, version uint64)
+	GetBytesLimit() int64
+	SetBytesLimit(bytesLimit int64)
+	BytesConsumed() int64
+	GetAll() []*statistics.Table
+	Clear()
+	Close()
+}
+
+const byteShards = 8
+const numShards = 1 << byteShards
+
+// statsCache caches Regions loaded from PD.
+type ristrettoStatsCache struct {
+	mu           sync.Mutex
+	cache        *cache.Cache
+	memCapacity  int64
+	version      uint64
+	memTracker   *memory.Tracker // track memory usage.
+	tablesShards [numShards]tableShard
+}
+
```

Comment:
Add a define `type tablesMap [numShards]tableShard`
And Set/Get/Del/GetAll() should be the member functions of `tablesMap`.